### PR TITLE
fix: random fixes

### PR DESCRIPTION
**Pull request by mosheavni**
random validations I added over time
change a bit the `not ready` in events
add `degraded` event

### CODE REVIEW

File: lua/kubectl/state.lua [hunk 1]
```diff
@@ -89,8 +89,8 @@ function M.checkVersions()
   commands.shell_command_async("kubectl", { "version", "--output", "json" }, function(data)
     local result = decode(data)
     if result then
-      local clientVersion = result.clientVersion.gitVersion
-      local serverVersion = result.serverVersion.gitVersion
+      local clientVersion = result and result.clientVersion and result.clientVersion.gitVersion or "0.0"
```
Comment (Ramilito) [inline lua/kubectl/state.lua:92]: Is the first check (result) needed? Don't we check that in the line above?

File: lua/kubectl/utils/events.lua [hunk 2]
```diff
@@ -68,13 +66,16 @@ function M.ColorStatus(status)
     AlreadyMountedVolume = true,
     Available = true,
     ContainerCreating = true,
+    Degraded = true,
```
Comment (Ramilito) [inline lua/kubectl/utils/events.lua:69]: The not ready ones I understand but degraded only warning?

File: lua/kubectl/views/nodes/definition.lua [hunk 3]
```diff
@@ -71,7 +71,7 @@ function M.getStatus(row)
   if ready and ready.status == "True" then
     return { symbol = hl.symbols.success, value = nodeConditions.NodeReady }
   end
-  return { symbol = events.ColorStatus("Error"), value = "Unknown" }
+  return { symbol = events.ColorStatus("NotReady"), value = "NotReady" }
```
Comment (Ramilito) [inline lua/kubectl/views/nodes/definition.lua:74]: I'm thinking we might keep the unknown as the last state and move the not ready into the if ready == true but in the elseif ready and ready.status == "false", what do you think?

File: lua/kubectl/state.lua (hunk 1 above)
Comment (mosheavni) [reply lua/kubectl/state.lua]: you are right

File: lua/kubectl/utils/events.lua (hunk 2 above)
Comment (mosheavni) [reply lua/kubectl/utils/events.lua]: sure thing, moved it to errorStatuses

File: lua/kubectl/views/nodes/definition.lua (hunk 3 above)
Comment (mosheavni) [reply lua/kubectl/views/nodes/definition.lua]: something like this?
[9cc3d80](https://github.com/Ramilito/kubectl.nvim/pull/490/commits/9cc3d809528c2ed5b92df63093e3054f1f66b425)

**Review by Ramilito — APPROVED** [review]

File: lua/kubectl/views/nodes/definition.lua (hunk 3 above)
Comment (Ramilito) [reply lua/kubectl/views/nodes/definition.lua]: Exactly like that! Would have written code example but was on phone 😅

**Pull request merged**